src/Type/Doctrine/Query/QueryResultTypeWalker.php

```diff
@@ -2188,4 +2195,18 @@ private function simpleFloatify(Type $type): Type
 		});
 	}
 
+	private function autodetectDbalTypePhpValue(DbalType $dbalType): Type
```

Comment:
Basically the same thing as ReflectionDescriptor does. So the deafult behaviour could be the same as registering ReflectionDescriptor for all registered types.